SAML2/XML/LibXML2.hsc

```diff
@@ -63,16 +65,55 @@ newtype Doc = Doc{ unDoc :: ForeignPtr XMLDoc }
 newDoc :: Ptr XMLDoc -> IO Doc
 newDoc = fmap Doc . newForeignPtr xmlFreeDoc
 
+data FromBytesError =
+  FromBytesResult
+    { fromBytesErr        :: Maybe IOError
+    , fromBytesResultNull :: Bool
+    , fromBytesNoise      :: String
+    , fromBytesInput      :: BS.ByteString
+    }
+  deriving (Eq, Show)
+
+instance Exception FromBytesError
+
+-- | Call libxml2 parser and return the parsed C object, or throw an error.
 fromBytes :: BS.ByteString -> IO Doc
 fromBytes s = do
-  d <- BSU.unsafeUseAsCStringLen s $ \(p, l) ->
-    throwErrnoIfNull "xmlReadMemory" $
-      xmlReadMemory p (fromIntegral l) nullPtr nullPtr (#{const XML_PARSE_NOENT} .|. #{const XML_PARSE_DTDLOAD} .|. #{const XML_PARSE_DTDATTR} .|. #{const XML_PARSE_NONET} .|. #{const XML_PARSE_COMPACT})
-  newDoc d
+  resetErrno
+  let config :: CInt
```

Comment:
Would be nice to have comments for those flags, because it's not obvious from the names what they do.

Reply:
it's not my code, and i only touched these to make it slightly easier to read.  also, i would argue this information belongs into the libxml2 manual.